.github/workflows/build_for_pypi.yml

```diff
@@ -1,46 +1,90 @@
+---
 name: Build and Optionally Publish to PyPi
 
 on:
-  workflow_call:
-    inputs:
-      publish:
-        type: boolean
-        default: false
-        description: "Publish to PyPi"
+  pull_request:
+    branches:
+      - main
+  push:
+    # Pattern matched against refs/tags
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
 
 jobs:
-  build_for_pypi:
+  build_dist_for_pypi:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-13, macos-latest]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          submodules: true
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.22.0
+        env:
+          CIBW_SKIP: pp*
+      - name: Store the distribution packages
+        uses: actions/upload-artifact@v4
+        with:
+          name: cibw-wheels-${{ matrix.os }}
+          path: ./wheelhouse/*.whl
+
+  build_src_for_pypi:
     runs-on: ubuntu-latest
-    permissions:
-      id-token: write # This is required for requesting the JWT
-      contents: read  # This is required for actions/checkout
     steps:
       - uses: actions/checkout@v4
         with:
+          persist-credentials: false
           submodules: true
       - name: Set up Python 3.11
         uses: actions/setup-python@v4
         with:
           python-version: "3.11"
-      - name: Install dependencies
+      - name: Install deps
         run: |
           pip install -r requirements.txt
-          python setup.py build
-          python setup.py develop
-      - name: Build distributions for different platforms
+          pip install build
+      - name: Build src dist
         run: |
-          pip install wheel
-          python setup.py sdist bdist_wheel --plat-name=manylinux2014_x86_64
-          python setup.py bdist_wheel --plat-name=macosx-12.6-universal2
-          python setup.py bdist_wheel --plat-name=win_amd64
-      - name: Move src file to match PEP625
+          python -m build --sdist
+      - name: Get version
+        id: version
         run: |
-          cd dist
-          ls | grep *.tar.gz | xargs -I x mv -- x "$(ls | grep *.tar.gz | sed 's/codecov-cli/codecov_cli/')"
-          ls -al
-          cd ..
+          version=$(python -c 'import importlib.metadata; print(importlib.metadata.version("codecov-cli"))')
+          echo "version=$version" >> "$GITHUB_OUTPUT"
```

Comment:
This output isn't used anywhere. If you need it, make sure to reference it in another step following this one, or map to a job output.
Otherwise, just drop the unused code:
```suggestion
```